packages/opentelemetry-test-utils/src/test-fixtures.ts

```diff
@@ -0,0 +1,251 @@
+    spanProcessor,
+    instrumentations: opts.instrumentations,
+  });
+  return sdk;
+}
+
+// TestSpan is an OTLP span plus references to `resource` and
+// `instrumentationScope` that are shared between multiple spans in the
+// protocol.
+export type TestSpan = ISpan & {
+  resource: IResource;
+  instrumentationScope: IInstrumentationScope;
+};
+
+/**
+ * A minimal HTTP server that can act as an OTLP HTTP/JSON protocol collector.
+ * It stores the received data for later test assertions.
+ *
+ * Limitations: This only supports traces so far, no metrics or logs.
+ * There is little error checking here; we are assuming valid OTLP.
+ */
+export class TestCollector {
+  endpointUrl?: string;
+  spans: Array<TestSpan> = [];
+  private _http;
+
+  constructor() {
+    this.clear();
+    this._http = createServer(this._onRequest.bind(this));
+  }
+
+  clear(): void {
+    this.spans = [];
+  }
+
+  // Start listening and set address to `endpointUrl`.
+  async start(): Promise<void> {
+    return new Promise(resolve => {
+      this._http.listen(() => {
+        this.endpointUrl = `http://localhost:${
+          (this._http.address() as AddressInfo).port
+        }`;
+        resolve();
+      });
+    });
+  }
+
+  close() {
+    this.endpointUrl = undefined;
+    return this._http.close();
+  }
+
+  // Return the spans sorted by start time for testing convenience.
+  get sortedSpans(): Array<TestSpan> {
+    return this.spans.slice().sort((a, b) => {
+      assert(typeof a.startTimeUnixNano === 'string');
+      assert(typeof b.startTimeUnixNano === 'string');
+      const aStartInt = BigInt(a.startTimeUnixNano);
+      const bStartInt = BigInt(b.startTimeUnixNano);
+      return aStartInt < bStartInt ? -1 : aStartInt > bStartInt ? 1 : 0;
+    });
+  }
+
+  _onRequest(req: IncomingMessage, res: ServerResponse) {
+    const parsedUrl = new URL(req.url as string, this.endpointUrl);
+    let instream: EventEmitter;
+    if (req.headers['content-encoding'] === 'gzip') {
+      instream = req.pipe(createGunzip());
+    } else {
+      req.setEncoding('utf8');
+      instream = req;
+    }
+
+    let body = '';
+    instream.on('data', (chunk: Buffer) => {
+      body += chunk;
+    });
+
+    instream.on('end', () => {
+      let resStatusCode;
+      const resHeaders = { 'content-type': 'application/json' };
+      let resBody = '';
+      if (req.method === 'POST' && parsedUrl.pathname === '/v1/traces') {
+        if (req.headers['content-type'] !== 'application/json') {
+          resStatusCode = 415;
+          resBody = JSON.stringify({ message: 'invalid content-type' });
+        } else {
+          this._ingestTraces(body);
+          resStatusCode = 200;
+          // TODO: I'm not sure what the ExportTraceServiceResponse form should be.
+          // https://github.com/open-telemetry/opentelemetry-proto/blob/v1.0.0/opentelemetry/proto/collector/trace/v1/trace_service.proto
+          resBody = '{}';
```

Comment:
Empty like this is fine. 🙂 

partialSuccess should only be set if there's certain spans set if some of the spans are rejected by the remote, we always accept everything here, so this is fine.

Reply:
Thanks for the confirmation. I removed the "TODO" in commit e0be467f.